lib/all_changed_attributes.rb

```diff
@@ -0,0 +1,28 @@
+module ArTransactionChanges
+  module AllChangedAttributes
+    def all_changed_attributes
+      @all_changed_attributes ||= ActiveSupport::HashWithIndifferentAccess.new
+    end
+
+    private
+
+    def store_to_all_changed_attributes(attr_name)
+      @all_changed_attributes = {attr_name.to_s => attributes[attr_name]}.merge(all_changed_attributes)
```

Comment:
This will create two ruby hash objects for each write.  In this case you just can modify the existing hash if it doesn't already have the key.